OSDOCS-18216-1: updates CoreDNS as GA (#106229)
openshift/openshift-docs: modules/con-about-using-onprem-dns-coredns.adoc

= About using on-premise DNS with CoreDNS

[role="_abstract"]
You can self-manage your on-premise DNS by integrating CoreDNS with your DNS infrastructure through a access control and zone delegation. {prodname} combines the DNS Operator with CoreDNS to simplify your management and security for on-premise DNS servers. You can use CoreDNS in both single-cluster and multi-cluster scenarios.

CoreDNS is best used in environments that change often, where using a DNS-as-code approach make sense. The following situations are example use cases for integrating with CoreDNS:

* You need to avoid dependency on external cloud DNS services.
* You have regulatory or compliance requirements mandating self-hosted infrastructure.
* You need to keep full control over DNS records.
* You want to delegate specific DNS zones from existing DNS servers to Kubernetes-managed CoreDNS.
* You require consistent DNS management across hybrid or multicloud environments.
* You need to reduce DNS operational costs by eliminating per-query charges.
* You do not want to directly manage DNS records on the on-premise DNS server.
* You need to keep authoritative control on edge DNS servers.

For example:

> * Configure your authoritative on-premise DNS server to delegate a specific subdomain, for example, `deployment.example.local`, to CoreDNS instances managed by {prodname}.

Reviewer: ```suggestion
* Configure your authoritative on-premise DNS server to delegate a specific subdomain, such as `deployment.example.local`, to CoreDNS instances managed by {prodname}.
```
This sounds better since For Example is used on line 23.

* Configure your authoritative on-premise DNS server to delegate a specific subdomain, for example, `deployment.example.local`, to CoreDNS instances managed by {prodname}.
* Any pod-level `dnsPolicy` can then interact with the CoreDNS provider within the {ocp} cluster. You can specific the DNS provider that handles the records for the targeted gateways in the `delete` field of the DNS policy.
* The CoreDNS instance becomes authoritative for the delegated subdomain and manages the necessary DNS records for gateways within that subdomain.

[id="rhcl-coredns-architecture_{context}"]
== CoreDNS integration architecture

CoreDNS is a DNS server that consists of default plugins that do several tasks, for example:

* Automatically detects when you add new services to your cluster and adds them to directories.
* Caches recent addresses to avoid the latency of repeated lookups.
* Can run health checks and skip over services that are down.
* Provides dynamic redirects by rewriting queries as they come in.

You can add plugins for observability and other services that you required by updating the CoreDNS Corefile.
//are there any limits on what we support in the corefile?

> //Q: does this come with RHCL, or do we need to install this plugin separately?

Reviewer 2: The plugin comes with our build of CoreDNS. We are investigating better ways to release CoreDNS at the moment, and I'll let you know once we change that. Right now they are [here](https://quay.io/repository/kuadrant/coredns-kuadrant?tab=tags&tag=latest) (build like [so](https://github.com/Kuadrant/dns-operator/actions/workflows/build-coredns-images.yaml)) and used [here](https://github.com/Kuadrant/dns-operator/blob/main/config/coredns/kustomization.yaml#L18).
